bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/TextLayout.java

```diff
@@ -2952,15 +2952,11 @@ StyleItem[] merge (long items, int itemCount) {
 			linkBefore = false;
 		}
 		char ch = segmentsText.charAt(start);
-		switch (ch) {
-			case '\r':
-			case '\n':
-				item.lineBreak = true;
-				break;
-			case '\t':
-				item.tab = true;
-				break;
+		if(ch == '\r' && (start+1 < end)) {
```

Comment:
The formatting of `if(` is poor.  The () around the < condition is redundant and odd that it's not used for the == test

Reply:
corrected it.